### PR TITLE
New resource - `azurerm_sentinel_data_connector_aws_s3`

### PR DESCRIPTION
**Pull request by magodo**
This PR adds support for a new sentinel data connector type: AWS S3. Checkout https://docs.microsoft.com/en-us/azure/sentinel/connect-aws for details.

As this resource needs setup of several AWS resources as a precondition, this PR didn't include an AccTest for it. I've locally tested the functionalities: CRUD and plan, which behaves as expected.

### CODE REVIEW

File: internal/services/sentinel/sentinel_data_connector_aws_s3_resource.go [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+
+func (r DataConnectorAwsS3Resource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+		"log_analytics_workspace_id": {
```
Comment (katbyte) [inline internal/services/sentinel/sentinel_data_connector_aws_s3_resource.go:40]:
```suggestion
		},
		
		"log_analytics_workspace_id": {
```

File: internal/services/sentinel/sentinel_data_connector_aws_s3_resource.go [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: loganalyticsValidate.LogAnalyticsWorkspaceID,
+		},
+
+		"aws_role_arn": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
```
Comment (katbyte) [inline internal/services/sentinel/sentinel_data_connector_aws_s3_resource.go:50]: can we validate this? maybe copy the validation function AWS provider uses

File: website/docs/r/sentinel_data_connector_aws_s3.html.markdown [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+The following arguments are supported:
+
+* `name` - (Required) The name which should be used for this AWS S3 Data Connector. Changing this forces a new AWS S3 Data Connector to be created.
+
+* `log_analytics_workspace_id` - (Required) The ID of the Log Analytics Workspace that this AWS S3 Data Connector resides in. Changing this forces a new AWS S3 Data Connector to be created.
+
+* `aws_role_arn` - (Required) The ARN of the AWS role, which is connected to this AWS CloudTrail Data Connector. See the [Azure document](https://docs.microsoft.com/en-us/azure/sentinel/connect-aws?tabs=s3#create-an-aws-assumed-role-and-grant-access-to-the-aws-sentinel-account) for details.
+
+* `destination_table` - (Required) The logs destination table name.
```
Comment (katbyte) [inline website/docs/r/sentinel_data_connector_aws_s3.html.markdown:69]: grammer

**Review by katbyte — CHANGES_REQUESTED** [review]

**Comment by magodo** [comment]
@katbyte Thank you for the comments! I've fixed all of them. Please take another look!

**Review by katbyte — APPROVED** [review]
LGTM 🔍

**Pull request merged**